docs/source/resources/graphql-glossary.md

```diff
@@ -0,0 +1,240 @@
+<p>A technique for caching inputs to GraphQL queries. This type of caching ensures that if the query is slightly different but with the same inputs, those inputs can simply be retrieved from the cache instead of fetching data again from the backend. It is implemented in Apollo Server 2 as Data Source caching.</p>
+
+<h2 id="query">Query</h2>
+<p>An operation that makes a GET request to a GraphQL service requesting for some data. It's better known as a read-only fetch operation.</p>
+
+<h2 id="query-colocation">Query colocation</h2>
+<p>A practice of placing a GraphQL query in the same location as the app component's view logic.</p>
+
+```js
+const GET_DOG_PHOTO = gql`
+ query dog($breed: String!) {
+  dog(breed: $breed) {
+    id
+    displayImage
+  }
+}`;
+
+export const queryComponent = `const DogPhoto = ({ breed }) => (
+  <Query query={GET_DOG_PHOTO} variables={{ breed }}>
+    {({ loading, error, data }) => {
+      if (loading) return null;
+      if (error) return 'Error!';
+      return (
+        <img src={data.dog.displayImage} />
+      );
+    }}
+  </Query>
+);`;
+```
+
+<h2 id="query-whitelisting">Query whitelisting</h2>
+<p>A technique for preventing unwanted attacks by maintaining a list of approved queries that are allowed in your application. Any query not present in the list that is run against the server will not be allowed. [Automatic Persisted Queries](../guides/performance.html#automatic-persisted-queries) is a tool by Apollo that enables query whitelisting and persisted queries.</p>
+
+<h2 id="resolver">Resolver</h2>
+<p>A function that connects schema fields and types to various backends. It can retrieve or write data from either an SQL, a No-SQL, graph database, a micro-service or a REST API.</p>
+
+
+<h2 id="schema">Schema</h2>
+<p>A model of the data that can be fetched from or written to a GraphQL server.</p>
+
+
+<h2 id="schema-definition-language">Schema Definition Language (SDL)</h2>
+<p>The syntax for writing GraphQL Schemas. It is otherwise known as Interface Definition Language. It is the lingua franca shared by all for building GraphQL APIs regardless of the programming language chosen.</p>
+
+
+<h2 id="schema-first-development">Schema first development</h2>
+<p>A development approach for designing and building modern UIs that involves the frontend and backend teams agreeing on a Schema first, which serves as a contract between the UI and the backend before any API engineering happens.</p>
+
+
+<h2 id="schema-registry">Schema registry</h2>
+<p>A central database that enables schema registration, tracking of detailed schema changes e.g. types added, fields added, fields deprecated and checking out previous versions of schema.</p>
+
+
+<h2 id="schema-versioning">Schema versioning</h2>
+<p>Refers to the ability to have different versions of your Schema. This allows for reversible changes to be made to the Schema. The Apollo CLI is a tool that provides and manages Schema versioning with Engine.</p>
```

Comment:
I'm not sure about this definition. The way we do it, we don't really focus on allowing multiple concurrent versions of a schema to be active at once. That'd require two different services on Engine.

Reply:
You mean in Engine, I can't see previous versions of a schema whenever I want? @JakeDawkins